src/service/document-info-impl.js

```diff
@@ -73,13 +78,31 @@ export class DocInfo {
           : sourceUrl;
     }
```

Comment:
The `canonicalUrl` in docInfo could be `rootNode.AMP.canonicalUrl` or `parseUrl(rootNode.querySelector('link[rel=canonical]').href).href` or `sourceUrl`. Shall I make the `linkRels['canonical']` the same logic?